presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1321,35 +1322,28 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
 
         private Scope getScopeFromTable(Table table, Optional<Scope> scope)
         {
-            QualifiedObjectName name = createQualifiedObjectName(session, table, table.getName());
-            Optional<TableHandle> tableHandle = session.getRuntimeStats().profileNanos(GET_TABLE_HANDLE_TIME_NANOS, () -> metadata.getTableHandle(session, name));
-            if (!tableHandle.isPresent()) {
-                if (!metadata.getCatalogHandle(session, name.getCatalogName()).isPresent()) {
-                    throw new SemanticException(MISSING_CATALOG, table, "Catalog %s does not exist", name.getCatalogName());
-                }
-                if (!metadata.schemaExists(session, new CatalogSchemaName(name.getCatalogName(), name.getSchemaName()))) {
-                    throw new SemanticException(MISSING_SCHEMA, table, "Schema %s does not exist", name.getSchemaName());
-                }
-                throw new SemanticException(MISSING_TABLE, table, "Table %s does not exist", name);
-            }
-
-            TableMetadata tableMetadata = session.getRuntimeStats().profileNanos(
-                    GET_TABLE_METADATA_TIME_NANOS,
-                    () -> metadata.getTableMetadata(session, tableHandle.get()));
```

Comment:
Is this some improvement or not by deleting this part of code?

Reply:
Code cleanup. MetadataResolver.getColumns already check for table existence, so no need to check twice.
The other option was to not really use metadata resolver method in this block.